x-pack/plugins/ml/public/application/app.tsx

```diff
@@ -50,11 +51,21 @@ export interface MlServicesContext {
 export type MlGlobalServices = ReturnType<typeof getMlGlobalServices>;
 
 const App: FC<AppProps> = ({ coreStart, deps, appMountParams }) => {
+  const redirectToMlAccessDeniedPage = async () => {
+    const accessDeniedPageUrl = await deps.share.urlGenerators
+      .getUrlGenerator(ML_APP_URL_GENERATOR)
+      .createUrl({
+        page: ML_PAGES.ACCESS_DENIED,
+      });
+    await coreStart.application.navigateToUrl(accessDeniedPageUrl);
+  };
+
   const pageDeps = {
     history: appMountParams.history,
     indexPatterns: deps.data.indexPatterns,
     config: coreStart.uiSettings!,
     setBreadcrumbs: coreStart.chrome!.setBreadcrumbs,
+    redirectToMlAccessDeniedPage,
```

Comment:
We now have two different approaches how this is accessible, one time it's used via hooks (`const redirectToMlAccessDeniedPage = useCreateAndNavigateToMlLink(ML_PAGES.ACCESS_DENIED);`) and the other approach is defined here via `deps`. I wonder if we have it available in `deps` anyway if we shouldn't use it via that all the time?

Reply:
Updated here https://github.com/elastic/kibana/pull/76735/commits/f762277c076b96176a2b7159cf577d6a6854b870